kafka-client/src/main/java/dev/responsive/kafka/api/stores/ResponsiveStores.java

```diff
@@ -116,10 +116,12 @@ public static KeyValueBytesStoreSupplier factStore(final String name) {
    *         that uses Responsive's storage for its backend
    */
   public static <K, V> StoreBuilder<KeyValueStore<K, V>> keyValueStoreBuilder(
-      final KeyValueBytesStoreSupplier storeSupplier,
+      final ResponsiveKeyValueBytesStoreSupplier storeSupplier,
       final Serde<K> keySerde,
       final Serde<V> valueSerde
   ) {
+    storeSupplier.params().setKeyAndValueSerdes(keySerde, valueSerde);
```

Comment:
@agavra it turns out we only have users pass in the serdes via the StoreBuilder, so we can only automatically grab serdes for them when they're using the PAPI

If we want to support row-level ttl for the DSL then we'll need to add these params either to the ResponsiveStores#Materialized or to the ResponsiveKeyValueParams itself. Would prefer not requiring it for the params personally but thoughts?

Reply:
😢 there was no other way to automatically get them? that's a real bummer - I feel like that's a worthwhile improvement to contribute to KS.

I think it's fine not requiring them, the API is already available in `ResponsiveKeyValueParams`, right? They could call `setKeyAndValueSerdes` themselves? We should just make sure to throw with a good exception message if they configure row level TTL and haven't passed in serdes.

Reply:
SG

And yeah, I agree we should just expose the serdes in Kafka Streams itself -- this should at least be a relatively short and easy KIP, whereas I can think of some hacky ways to expose the original deserialized key & value but it would probably be a longer & more difficult KIP discussion for that

Reply:
If we are going to have to require the key and value serdes for key/value based TtlProviders, then rather than adding them as optional (and easily missable) params in the ResponsiveKeyValueParams, we might as well just add these to the `.withKey/ValueBased` builder methods for TtlProvider. I'll make this change but lmk if you prefer otherwise or have a different idea

Reply:
> whereas I can think of some hacky ways to expose the original deserialized key & value

ah this is what I was hoping the KIP would do... (not expose the Serdes directly)

> just add these to the .withKey/ValueBased builder methods for TtlProvider.

yeah let's just do that

Reply:
> ah this is what I was hoping the KIP would do... (not expose the Serdes directly)

ah I shouldn't discourage you from trying this, I guess it might be possible to add the deserialized key/value to the StateStoreContext? I'm not entirely sure this is possible but I didn't put much thought into this so I wouldn't say I'm sure it's impossible either

Reply:
Actually I don't know what I was talking about, it should be pretty straightforward. Just add APIs for deserialized key/value to the StateStoreContext and then we can set these in the store layer right before we serialize them (which IIRC is the MeteredXYZStore layer)